src/algorithms/calorimetry/ImagingTopoClusterConfig.h

```diff
@@ -7,14 +7,20 @@
 
 namespace eicrecon {
 
+  enum layerMode_enum {etaphi, xy};
   struct ImagingTopoClusterConfig {
 
     // maximum difference in layer numbers that can be considered as neighbours
     int neighbourLayersRange = 1;
     // maximum distance of local (x, y) to be considered as neighbors at the same layer
     std::vector<double> localDistXY = {1.0 * dd4hep::mm, 1.0 * dd4hep::mm};
-    // maximum distance of global (eta, phi) to be considered as neighbors at different layers
+    // maximum distance of global (eta, phi) to be considered as neighbors at different layers (if layerMode=="etaphi")
     std::vector<double> layerDistEtaPhi = {0.01, 0.01};
+    // maximum distance of global (x, y) to be considered as neighbors at different layers (if layerMode=="xy")
+    std::vector<double> layerDistXY = {1.0 * dd4hep::mm, 1.0 * dd4hep::mm};
+    // determines how neighbors are determined for hits in different layers (using either eta and phi, or x and y)
+    std::string layerMode ="etaphi";
```

Comment:
The idea here is to use the enum in the way the mode is specified, not a string (with run-time validity checks).
```suggestion
    enum ELayerMode {etaphi, xy} layerMode = etaphi;
```
e.g. https://godbolt.org/z/KWWWTc63W.

Reply:
I just checked in this change and a few others to try to get this to work (and have checked those in).   I get the error below.  If you can get this to work, let me know.  Otherwise I will revert these changes and use a string in config like I was doing before.  

In file included from /usr/local/include/JANA/JApplication.h:26,
                 from /Users/spaul/EIC/EICrecon/src/detectors/ZDC/ZDC.cc:7:
/usr/local/include/JANA/Services/JParameterManager.h: In instantiation of ‘static void JParameterManager::Parse(const std::string&, T&) [with T = eicrecon::ImagingTopoClusterConfig::ELayerMode; std::string = std::__cxx11::basic_string<char>]’:
/Users/spaul/EIC/EICrecon/src/extensions/jana/JOmniFactory.h:293:45:   required from ‘void JOmniFactory<AlgoT, ConfigT>::ParameterRef<T>::Configure(std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >) [with T = eicrecon::ImagingTopoClusterConfig::ELayerMode; AlgoT = eicrecon::ImagingTopoCluster_factory; ConfigT = eicrecon::ImagingTopoClusterConfig]’
/Users/spaul/EIC/EICrecon/src/extensions/jana/JOmniFactory.h:284:14:   required from here
/usr/local/include/JANA/Services/JParameterManager.h:322:19: error: static assertion failed: Type is not automatically parseable by JParameterManager. To use, provide a template specialization for JParameterManager::Parse(std::string in, T& out).
  322 |     static_assert(parseable, "Type is not automatically parseable by JParameterManager. To use, provide a template specialization for JParameterManager::Parse(std::string in, T& out).");
      |                   ^~~~~~~~~
/usr/local/include/JANA/Services/JParameterManager.h:322:19: note: ‘parseable’ evaluates to false
gmake[2]: *** [src/detectors/ZDC/CMakeFiles/ZDC_plugin.dir/build.make:76: src/detectors/ZDC/CMakeFiles/ZDC_plugin.dir/ZDC.cc.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:2394: src/detectors/ZDC/CMakeFiles/ZDC_plugin.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....

Reply:
JParameterManager does not have default string interpretation for the enum. For now, either just disable (comment with `FIXME` is ok) the ParameterRef for layerMode, or implement the parsing functionality as it indicates.

The simplest approach (which doesn't require JANA2's `JParameterManager::Parse(const std::string in, T& out)` suggestion) may be something like this in your config header:
```cpp
std::istream& operator>>(std::istream& in, ELayerMode& layerMode)
{
  std::string s{};
  in >> s;
  if (s == "etaphi")
    return ELayerMode::etaphi;
  if (s == "xy")
    return ELayerMode::xy;
  return {};
}
```

Reply:
Note that disabling the ParameterRef will mean the layer mode won't be settable on the command line, only through the plugin configuration files in C++, but I think in this case that's likely sufficient so I don't feel that this suggestion is restrictive.

Reply:
If you want bonus points, then a way to simultaneously define the enum and their string representation would be nice... Something with multimap or so.

Reply:
^^^
Using a map or multimap would also let you easily check if the user passed a known mode! (We do something similar in the jet reconstruction algorithm...)
```
  try {
    m_mapJetAlgo.at(m_cfg.jetAlgo);
  } catch (std::out_of_range& out) {
    this->error(" Unknown jet algorithm \"{}\" specified!", m_cfg.jetAlgo);
    throw JException(out.what());
  }
```

Reply:
It appears to work now; not the most elegant solution, but it does compile and it grabs the value of the layerMode from the src/detectors/ZDC/ZDC.cc file (and the equivalent for FHCAL).  If you are satisfied with this, let me know.